plutus-scb/src/Cardano/Metadata/Server.hs

```diff
@@ -39,56 +42,48 @@ import qualified Network.Wai.Handler.Warp        as Warp
 import           Plutus.SCB.App                  (App)
 import           Plutus.SCB.SCBLogMsg            (ContractExeLogMsg (StartingMetadataServer))
 import           Plutus.SCB.Utils                (tshow)
-import           Servant                         (Application, Handler (Handler), ServerError, err404, errBody,
+import           Servant                         (Application, Handler (Handler), ServerError, err500, errBody,
                                                   hoistServer, serve)
 import           Servant.API                     ((:<|>) ((:<|>)))
 import           Servant.Client                  (baseUrlPort)
 import           Wallet.Emulator.Wallet          (Wallet (Wallet), walletPrivKey, walletPubKey)
 
 toServerError :: MetadataError -> ServerError
-toServerError (SubjectNotFound subject) =
-    err404 {errBody = BSL8.pack $ "Subject not found: " <> show subject}
-toServerError (PropertyNotFound subject propertyKey) =
-    err404
+toServerError (MetadataClientError err) =
+    err500
         { errBody =
-              BSL8.pack $
-              "Property not found: " <> show subject <> ", " <> show propertyKey
+              BSL8.fromStrict . encodeUtf8 $
+              "Error contacting metadata server: " <> tshow err
         }
 
 fetchSubject ::
-       ( Member (LogMsg MetadataLogMessage) effs
-       , Member (Error MetadataError) effs
-       )
+       Member (LogMsg MetadataLogMessage) effs
     => Subject
-    -> Eff effs [Property]
+    -> Eff effs (Maybe (SubjectProperties 'AesonEncoding))
 fetchSubject subject = do
     logInfo $ FetchingSubject subject
-    let matches = filter (\v -> _propertySubject v == subject) testDatabase
-    case matches of
-        [] -> throwError $ SubjectNotFound subject
-        _  -> pure matches
+    case Map.lookup subject testDatabase of
+        Nothing    -> pure Nothing
+        Just match -> pure $ Just $ SubjectProperties subject match
```

Comment:
micro-comment: I'm a fan of the hlint thing `pure . Just $ ...`

Reply:
also, why not use `fmap` here?

Reply:
and you are lifting a pure function into `Eff` just for the sake of logging, seems to me that it should be a pure function and you add the logging where it gets called.

Reply:
Discussed offline and addressed. Thanks!